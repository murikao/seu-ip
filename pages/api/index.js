export default function handler(req, res) {
  
  res.statusCode = 200
  var ip = req.headers['x-forwarded-for'] || 
     req.connection.remoteAddress || 
     req.socket.remoteAddress ||
     (req.connection.socket ? req.connection.socket.remoteAddress : null);
     ip = ip.match(/\d{1,3}\.\d{1,3}\.\d{1,3}.\d{1,3}/)
  console.log(">>>>>>>api/"+ip)
  res.setHeader('Content-Type', 'application/json')
  res.end(JSON.stringify({ "ip":ip[0]}))
  //res.send(req.toString())
}
