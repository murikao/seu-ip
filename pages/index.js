import React  from "react";
import Image from 'next/image'




 

var imgX = require('../images/ip.png');
//import Module from "module"
//var Fs = require("fs")

/* Module._extensions['.png'] = function(module, fn) {
  var base64 = Fs.readFileSync(fn).toString('base64');
  module._compile('module.exports="data:image/jpg;base64,' + base64 + '"', fn);
};
 */
//const imgIp = require( "../images/ip.png")




export async function getServerSideProps({req}) {
  // Fetch data from external API
  var ip = req.headers['x-forwarded-for'] || 
     req.connection.remoteAddress || 
     req.socket.remoteAddress ||
     (req.connection.socket ? req.connection.socket.remoteAddress : null);
  console.log(">>>>>>>"+ip.match(/\d{1,3}\.\d{1,3}\.\d{1,3}.\d{1,3}/))
  const ip4=ip.toString().match(/192\.168/)=="192.168"?"":ip
  const res = await fetch(`https://king.host/wiki/wp-content/themes/kinghost-wiki/includes/ip-api.php?ip=${ip4}`)
  const data = await res.json()

  // Pass data to the page via props
  return { props: {data}  }
}





var geoLocal = {
  status: "success",
  country: "Brazil",
  countryCode: "BR",
  region: "MG",
  regionName: "Minas Gerais",
  city: "Belo Horizonte",
  isp: "Claro S.A."
  
}


function home({data}) {
  
  var ip =""
   const geo = data
  var tela = (<div>Tela de retorno</div>)
   //localizacao(ip).then( geoLocal => { setGeo(geoLocal)
    console.log(geo.status)

     return  (
  <div style={{alignItems: "center",
            display: "flex",
            flexDirection: "column",
            flexWrap: "wrap",
            justifyContent: "center"}}>
                        <Image  src={imgX} alt="ip client" width={100} height={120}/>
             <h1 id="centro">Seu ip: <font color="red">{geo.query}</font></h1>
   
             <br/>
             <br/>
             <h2>
             Localização:<font color="red">  {geo.city}, {geo.region}, {geo.country} - ({geo.countryCode})</font>
             <br/>
             <br/>
             Operadora : <font color="red"> {geo.isp}</font></h2>
             <br/>
             <br/>
             <br/>
             <h4>  Desenvolvido por ...
                   <strong> <font color="red"> Murikão</font>
                   </strong>
              </h4>
          </div>)
  } 

  


export default home

