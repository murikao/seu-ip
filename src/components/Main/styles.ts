import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Logo = styled.img`
  margin-bottom: 2rem;
  width: min(10rem, 50%);
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
  margin-bottom: 2rem;
`

export const Illustration = styled.img`
  margin-top: 3rem;
  width: max(30%, 15rem);
  height: min(15rem, 70%);
`
export const IpNumber = styled.span`
  font-size: 3rem;
  color: yellow;
  font-weight: 400;
  margin-bottom: 2rem;
`
export const IpInfo = styled.span`
  font-size: 2rem;
  color: yellow;
  font-weight: 400;
  margin-bottom: 2rem;
`
