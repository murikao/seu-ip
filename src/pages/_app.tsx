/* // MyApp do RocketSeat Boilerplate

//////

const Boilerplate =`Em programação de computadores, código boilerplate ou boilerplate se refere a 
 seções de código que devem ser incluídas em muitos lugares com pouca ou nenhuma
 alteração. Ele é muitas vezes usado quando se refere a linguagens que são
 consideradas detalhadas, onde o programador deve escrever muito código para fazer
 tarefas mínimas. 
`
//////
import React from 'react'
import { AppProps } from 'next/app'
import { ThemeProvider } from 'styled-components'

import GlobalStyle from '../styles/global'
import theme from '../styles/theme'

const MyApp: React.FC<AppProps> = ({ Component, pageProps }) => {
  return (
    <ThemeProvider theme={theme}>
      <Component {...pageProps} />
      <GlobalStyle />
    </ThemeProvider>
  )
}

export default MyApp */

import { AppProps } from 'next/app'
import Head from 'next/head'

import GlobalStyles from '../styles/global'

function App({ Component, pageProps }: AppProps) {
  return (
    <>
      <Head>
        <title>React Avançado - Boilerplate</title>
        <link rel="shortcut icon" href="/img/icon-512.png" />
        <link rel="apple-touch-icon" href="/img/icon-512.png" />
        <link rel="manifest" href="/manifest.json" />
        <meta
          name="description"
          content="A simple project starter to work with TypeScript, React, NextJS and Styled Components"
        />
      </Head>
      <GlobalStyles />
      <Component {...pageProps} />
    </>
  )
}

export default App
