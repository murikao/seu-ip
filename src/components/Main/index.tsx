import * as S from './styles'

const Main = ({
  data = {
    status: 'status',
    country: 'country',
    countryCode: 'XX',
    region: 'XX',
    regionName: 'state',
    city: 'city',
    isp: 'Xxxx S.A.',
    query: '127.0.0.0'
  }
}) => {
  const title = 'Seu IP'
  const description = 'Abaixo algumas informações do seu endereço IP'

  const geo = data
  return (
    <S.Wrapper>
      <S.Logo
        src="/img/ip.svg"
        alt="Imagem de localizador tipo google maps azul escrito IP dentro."
      />
      <S.Title>{title}</S.Title>
      <S.IpNumber> {geo.query}</S.IpNumber>
      <S.Description>
        {description}
        <br />
        <br />
        Localização:
        <S.IpInfo>
          {' '}
          {geo.city}, {geo.region}, {geo.country} - ({geo.countryCode})
        </S.IpInfo>
        <br />
        <br />
        Operadora : <S.IpInfo>{geo.isp}</S.IpInfo>
      </S.Description>
      <S.Illustration
        src="/img/worldIp.jpg"
        alt="Um mapa do mundo com linhas ligando os endereços de ip"
      />
    </S.Wrapper>
  )
}
export default Main
