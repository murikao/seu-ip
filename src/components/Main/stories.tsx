import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    data: {
      status: 'success',
      country: 'Brazil',
      countryCode: 'BR',
      region: 'MG',
      regionName: 'Minas Gerais',
      city: 'Belo Horizonte',
      isp: 'Claro S.A.',
      query: '192.168.0.82'
    },
    title: 'title default',
    description: 'description default'
  }
} as Meta

export const Basic: Story = args => <Main {...args} />
Basic.args = {
  title: 'title basic',
  description: 'description basic'
}

export const Default: Story = args => <Main {...args} />
