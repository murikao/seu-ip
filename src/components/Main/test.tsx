import { render, screen } from '@testing-library/react'
import { ipRes } from 'pages'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(screen.getByRole('heading', { name: /Seu IP/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('Page', () => {
  it('should get ip info from url', async () => {
    /* const res = await fetch(
    `https://king.host/wiki/wp-content/themes/kinghost-wiki/includes/ip-api.php?ip=`
  )
  const dataProp = await res.json() */
    const dataip = await (await ipRes('')).json()
    expect(dataip.status).toBe('success')
  })
})
