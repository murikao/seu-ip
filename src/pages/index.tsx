import Main from 'components/Main'
import { GetServerSideProps, NextPage } from 'next'
import fetch from 'node-fetch'

interface Props {
  data: {
    status: string
    country: string
    countryCode: string
    region: string
    regionName: string
    city: string
    isp: string
    query: string
  }
}
export const ipRes = async (ip4: string) =>
  await fetch(
    `https://king.host/wiki/wp-content/themes/kinghost-wiki/includes/ip-api.php?ip=${ip4}`
  )

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  // Fetch data from external API

  const { remoteAddress } = req.connection
  const socketIP = req.socket.remoteAddress
  const ip = remoteAddress || socketIP

  const localIp = /192\.168|^127\./
  const ipValido = /\d{1,3}\.\d{1,3}\.\d{1,3}.\d{1,3}/g
  const _ip = ip ? (ipValido.test(ip) ? ip.match(ipValido) : '') : ''
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let ip4: any
  if (_ip && Object.prototype.toString.call(_ip) === '[object Array]') {
    ip4 = _ip[0]
  } else {
    if (_ip != null) {
      ip4 = _ip
    }
  }
  ip4 = localIp.test(ip4) ? '' : ip4

  /*  const res = await fetch(
    `https://king.host/wiki/wp-content/themes/kinghost-wiki/includes/ip-api.php?ip=${ip4}`
  )
  const dataProp = await res.json()
 */
  const dataip = await (await ipRes(ip4)).json()

  console.log(new Date())
  // console.log(dataProp)
  console.log('dataip')
  console.log(dataip)

  // Pass data to the page via props
  return { props: { data: dataip } }
}

const Home: NextPage<Props> = ({ data }) => {
  return <Main data={data} />
}
export default Home
