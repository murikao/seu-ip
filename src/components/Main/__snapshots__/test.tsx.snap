// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin-bottom: 2rem;
  width: min(10rem,50%);
}

.c2 {
  font-size: 2.5rem;
}

.c4 {
  font-size: 2rem;
  font-weight: 400;
  margin-bottom: 2rem;
}

.c6 {
  margin-top: 3rem;
  width: max(30%,15rem);
  height: min(15rem,70%);
}

.c3 {
  font-size: 3rem;
  color: yellow;
  font-weight: 400;
  margin-bottom: 2rem;
}

.c5 {
  font-size: 2rem;
  color: yellow;
  font-weight: 400;
  margin-bottom: 2rem;
}

<main
  class="c0"
>
  <img
    alt="Imagem de localizador tipo google maps azul escrito IP dentro."
    class="c1"
    src="/img/ip.svg"
  />
  <h1
    class="c2"
  >
    Seu IP
  </h1>
  <span
    class="c3"
  >
     
    127.0.0.0
  </span>
  <h2
    class="c4"
  >
    Abaixo algumas informações do seu endereço IP
    <br />
    <br />
    Localização:
    <span
      class="c5"
    >
       
      city
      , 
      XX
      , 
      country
       - (
      XX
      )
    </span>
    <br />
    <br />
    Operadora : 
    <span
      class="c5"
    >
      Xxxx S.A.
    </span>
  </h2>
  <img
    alt="Um mapa do mundo com linhas ligando os endereços de ip"
    class="c6"
    src="/img/worldIp.jpg"
  />
</main>
`;
